libs/node/src/services/node-crypto-function.service.ts

```diff
@@ -172,24 +173,33 @@ export class NodeCryptoFunctionService implements CryptoFunctionService {
     mac: string | null,
     key: SymmetricCryptoKey,
   ): CbcDecryptParameters<Uint8Array> {
-    const p = {} as CbcDecryptParameters<Uint8Array>;
-    p.encKey = key.encKey;
-    p.data = Utils.fromB64ToArray(data);
-    p.iv = Utils.fromB64ToArray(iv);
-
-    const macData = new Uint8Array(p.iv.byteLength + p.data.byteLength);
-    macData.set(new Uint8Array(p.iv), 0);
-    macData.set(new Uint8Array(p.data), p.iv.byteLength);
-    p.macData = macData;
-
-    if (key.macKey != null) {
-      p.macKey = key.macKey;
-    }
-    if (mac != null) {
-      p.mac = Utils.fromB64ToArray(mac);
+    const dataBytes = Utils.fromB64ToArray(data);
+    const ivBytes = Utils.fromB64ToArray(iv);
+    const macBytes = mac != null ? Utils.fromB64ToArray(mac) : null;
+
+    const innerKey = key.inner();
+
+    if (innerKey.type === EncryptionType.AesCbc256_B64) {
+      return {
+        iv: ivBytes,
+        data: dataBytes,
+        encKey: innerKey.encryptionKey,
+      } as CbcDecryptParameters<Uint8Array>;
+    } else if (innerKey.type === EncryptionType.AesCbc256_HmacSha256_B64) {
+      const macData = new Uint8Array(ivBytes.length + dataBytes.length);
+      macData.set(ivBytes, 0);
+      macData.set(dataBytes, ivBytes.length);
```

Comment:
```suggestion
      macData.set(new Uint8Array(ivBytes), 0);
      macData.set(new Uint8Array(dataBytes), ivBytes.length);
```
💭  Do we need to worry about any pass by reference/ object mutation happening, and why these were passed in new arrays in the past? Looks like we have this pattern else where in the code base.

Reply:
We should never mutate keys or encstrings, and I'm not aware of us doing mutation. I'm not sure why this is the case in other parts of the codebase.

Also, I'm not quite sure I follow this proposed change. `ivBytes` and `dataBytes` are copied by `macData.set`. So I'm not sure what creating another `Uint8Array` does here? Is it not just an extra copy (that we then copy from)?

Reply:
What benefit does rewriting this get us? The risk is a potential unknown and splitting the code base on the pattern.

For this specifically, if you feel strongly on going this path I'm good approving after doing some testing and reading the docs.

Reply:
I don't feel too strongly about this, since this code is in the medium term replaced by SDK entirely. I've added back the `new Uint8Array()` to the above mentioned section of code.